packit/api.py

```diff
@@ -286,9 +286,11 @@ def sync_release(
             if not use_local_content:
                 self.up.local_project.git_repo.git.checkout(current_up_branch)
             self.dg.refresh_specfile()
-            branch = local_pr_branch if create_pr else dist_git_branch
+            remote_and_branch = (
+                f"fork/{local_pr_branch}" if create_pr else f"origin/{dist_git_branch}"
+            )
             self.dg.local_project.git_repo.git.reset(
-                "--hard", f"remotes/origin/{branch}"
+                "--hard", f"remotes/{remote_and_branch}"
```

Comment:
What about using HEAD as I've proposed last time?

We are checking out the branch at the beginning so only not-committed changes can harm anything.

Reply:
Ok, good idea and requre tests also pass :+1:  
Changed